lib/compat/wordpress-6.4/block-hooks.php

```diff
@@ -73,6 +73,7 @@ function gutenberg_add_hooked_blocks( $settings, $metadata ) {
 		'keywords'         => 'keywords',
 		'example'          => 'example',
 		'variations'       => 'variations',
+		'allowed_blocks'   => 'allowedBlocks',
```

Comment:
I think it's fine to add here. This is the compatibility layer that implements the Block Hooks "shim" for when the GB plugin runs on WP < 6.4; we're [adding the `blockHooks` field](https://github.com/WordPress/gutenberg/pull/58262/files#diff-0019d64feee50979366751fa8f69fda30459ebfa3e5fc2b15ee295feae284c78R79) to the preloaded block settings (which unfortunately involves duplicating [the `$fields_to_pick` allowlist from Core](https://github.com/WordPress/wordpress-develop/blob/2ce87aa6f530dc3a80780f8c3b65f5f17d3d1321/src/wp-admin/includes/post.php#L2287-L2306), as it's not exposed by any filter or the like).

Since we're now also adding the `allowed_blocks` field in `block.json`, it makes sense to include that field in the allowlist 👍

Reply:
I don't think we should update files under `6.4` compat folder. If we intend to use this feature in `6.5`, we should have a new file under that folder and I guess it will require a backport.  --cc @gziolo @Mamaduka

Reply:
Once we have the WordPress Core part as outlined in https://github.com/WordPress/gutenberg/pull/58262#issuecomment-1914493617, then it definitely might become an issue when support for WordPress 6.3 in the Gutenberg plugin drops and this polyfill is no longer there.

Reply:
Nik is right. PHP code should be moved to a target WP version compat directory to avoid missing the backport.

Reply:
Here I modified `gutenberg_add_hooked_blocks` function that's intended to polyfill the block hooks behavior for WP < 6.4. WordPress 6.4 and newer already has the relevant code directly in the `register_block_type_from_metadata` function and doesn't need any patching.

I'll need to add a separate WordPress 6.5 back compat code in order to support `allowedBlocks`, both in Core and in the Gutenberg plugin. But I haven't done that yet.

Reply:
@gziolo @ockham https://core.trac.wordpress.org/ticket/60403 and https://github.com/WordPress/wordpress-develop/pull/5988 should be all we need on Core to support `allowedBlocks`. I closely followed what Bernie did when adding `blockHooks`.

Reply:
Excellent, I’ll take it from here and commit tomorrow unless Bernie is faster. Thank you so much @jsnajdr. Stellar job!